### ISSUE TITLE
Testing PkgPages

### ISSUE THREAD

**Issue by shashi** [tlienart/Franklin.jl#578]
I often find that some API change breaks the docs and I don't know about it until someone points it out with a screenshot.

It would be nice if package tests to also automatically run `optimize` and failed if there are any errors in running the code blocks in the docs, except those I mark with a comment such as `# ok-to-error`...

This is a fancy feature request, not super important but would be lovely to have!

Ideally I should be able to just add:

```julia
using PkgPage
testpage()
```

to the end of my `test/runtests.jl` (possibly within `if haskey(ENV, "TRAVIS")`) and have it run.

**Comment by tlienart** [comment]
transferring to Franklin as it's specific to it

**Comment by shashi** [comment]
Ah okay, thanks! I want to run it for my Julia package documentation which ships with the code. But I guess it could work for static sites as well :)

**Comment by tlienart** [comment]
yes but PkgPage is basically just a template for Franklin and what you asked for is definitely something that would be offered at Franklin-level :)
